crates/ruff_diagnostics/src/fix.rs

```diff
@@ -27,11 +27,13 @@ pub enum Applicability {
     Unspecified,
 }
 
-/// Indicates whether a fix should be applied in isolation or as part of a batch.
+/// Indicates the isolation constraints for a fix.
 #[derive(Default, Copy, Clone, Debug, Hash, PartialEq, Eq, PartialOrd, Ord)]
 #[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
 pub enum Isolation {
-    Isolated,
+    /// At most one fix from this isolation group should be applied per run.
+    Group(u32),
```

Comment:
How do we make sure that the group ids remain unique? I'm worried that this becomes a `z-index` like problem where people start guessing numbers. 

 I think we could either restrict the type to `NodeId` if the idea is to isolate operations on a per node or we could consider using a `TypeId` for fixed groups (probably less useful)

Reply:
The problem is, this crate doesn't depend on `ruff_python_semantic` :(

Reply:
I agree with you, but I can't figure out how to fix this.

Reply:
I'm going to bias towards merging, but if you have suggestions I'm more than happy to act on them.